icons.go

```diff
@@ -76,7 +87,11 @@ func (im iconMap) get(f *file) string {
 	case f.Mode().IsRegular() && f.Mode()&0111 != 0:
 		key = "ex"
 	default:
-		key = "*" + filepath.Ext(f.Name())
+		if (gOpts.globicon) {
+			key = im.findkey(f.Name())
+		} else {
+			key = "*" + filepath.Ext(f.Name())
+		}
```

Comment:
We can combine this section and `findKey` into `iconKeyGenerators` as:

```go
func(f *file) (string, bool) {
    if (!gOpts.globicon) {
        goto cancel
    }

    for key,_ := range(im) {
        match, err := filepath.Match(key, filename)
        if err == nil && match {
            return key
        }
    }

cancel:
    return "", false
}
```

There's a downside here though. Really we should try to cancel out any remaining
glob checks if this function returns false. So perhaps @SPFabGerman is right and
we should create a `iconmatchingorder` option.